### PR TITLE
merge Dev

### PR DESCRIPTION
**Pull request by aobo-y**
Hi @jasonjabbour @kaiyingshan 

Do you guys think we can merge the dev into master and deploy to the laptops? I think Jason have finished all the complicated logic and Kaiying's testing does not impact the system. Please let me know if there are any concerns~

### CODE REVIEW

File: pkg/recommender.py [hunk 1]
```diff
@@ -83,9 +93,22 @@ def update(self, *args, **kargs):
 
 
 class Recommender:
-    def __init__(self, evt_dim=5, mock=False, server_config=temp_server_config, mode='default'):
+    def __init__(self, evt_dim=5, mock=False, server_config=temp_server_config, 
+    mode='default', test=False, time_config=None, schedule_evt_test_config=None):
```
Comment (aobo-y) [inline pkg/recommender.py:97]: I feel as long as the interface of the constructor and function `dispatch` stays the same, we do not need to inform acoustic side to have any updates. Acoustic system should use all these default arguments, right?

Comment (jasonjabbour) [reply pkg/recommender.py:97]: Oh okay and yes I believe so

**Comment by jasonjabbour** [comment]
Hi Aobo, I pushed a view minor changes to my branch last night. It doesn't seem like this pull request is up to date with my code. Should I make a new pull request?

**Comment by aobo-y** [comment]
@jasonjabbour sure, just create a PR to the dev. after it is merged into dev, ur commit will appear here

**Comment by aobo-y** [comment]
FYI, @jasonjabbour @kaiyingshan i will merge this into master and inform Ashley to pull

**Pull request merged**